src/lib/assets/yaml.ts

```diff
@@ -24,174 +26,154 @@ export async function overridesFile({ hash, name, image, config, apiToken }: Ass
-      },
-      readinessProbe: {
-        httpGet: {
-          path: "/healthz",
-          port: 3000,
-          scheme: "HTTPS",
-        },
-        initialDelaySeconds: 10,
-      },
-      livenessProbe: {
-        httpGet: {
-          path: "/healthz",
-          port: 3000,
-          scheme: "HTTPS",
-        },
-        initialDelaySeconds: 10,
-      },
-      resources: {
-        requests: {
-          memory: "64Mi",
-          cpu: "100m",
-        },
-        limits: {
-          memory: "256Mi",
-          cpu: "500m",
-        },
-      },
-      containerSecurityContext: {
-        runAsUser: 65532,
-        runAsGroup: 65532,
-        runAsNonRoot: true,
-        allowPrivilegeEscalation: false,
-        capabilities: {
-          drop: ["ALL"],
-        },
-      },
-      nodeSelector: {},
-      tolerations: [],
-      extraVolumeMounts: [],
-      extraVolumes: [],
-      affinity: {},
-      podAnnotations: {},
-      serviceMonitor: {
-        enabled: false,
-        labels: {},
-        annotations: {},
+      name,
+    ),
+    watcher: generateWatcherConfig(
+      {
+        ...config,
+        webhookTimeout: config.webhookTimeout ?? 30, // Provide a default value if undefined
+        alwaysIgnore: { namespaces: config.alwaysIgnore?.namespaces ?? [] }, // Ensure alwaysIgnore.namespaces is always an array
       },
+      image,
+      name,
+    ),
+  };
+}
+
+// Generates the admission object configuration
+interface Config {
+  onError?: string;
+  webhookTimeout: number;
+  description?: string;
+  uuid: string;
+  alwaysIgnore: {
+    namespaces: string[];
+  };
+  [key: string]: unknown; // Add other properties as needed
+}
+
+export function generateAdmissionConfig(config: Config, image: string, name: string) {
+  if (!config.uuid) {
+    throw new Error("uuid is required in config");
+  }
+
+  return {
+    terminationGracePeriodSeconds: 5,
+    failurePolicy: config.onError === "reject" ? "Fail" : "Ignore",
+    webhookTimeout: config.webhookTimeout ?? 30, // Provide a default value
+    env: genEnv(config, false, true),
+    image,
+    annotations: { "pepr.dev/description": config.description || "" },
+    labels: {
+      app: name,
+      "pepr.dev/controller": "admission",
+      "pepr.dev/uuid": config.uuid,
     },
+    securityContext: generateSecurityContext(),
+    readinessProbe: generateProbeConfig(),
+    livenessProbe: generateProbeConfig(),
+    resources: generateResourceConfig(),
+    containerSecurityContext: generateContainerSecurityContext(),
+    nodeSelector: {},
+    tolerations: [],
+    extraVolumeMounts: [],
+    extraVolumes: [],
+    affinity: {},
+    podAnnotations: {},
+    serviceMonitor: { enabled: false, labels: {}, annotations: {} },
   };
+}
 
-  await fs.writeFile(path, dumpYaml(overrides, { noRefs: true, forceQuotes: true }));
+// Generates the watcher object configuration
+export function generateWatcherConfig(config: Config, image: string, name: string) {
```

Comment:
```suggestion
export function generateWatcherConfig(config: Config, image: string, name: string) {
```

Function name is clear, could use a return type

Reply:
Done